test/setup.js

```diff
@@ -12,36 +12,51 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-import should from 'should';
 import sinon from 'sinon';
-import 'should-sinon';
+import chai, { Assertion } from 'chai';
+import 'chai/register-should';
+import chaiAsPromised from 'chai-as-promised';
+import sinonChai from 'sinon-chai';
 import naclFactory from 'js-nacl';
 
 process.env.NODE_ENV = 'test';
 
-should.use((_, Assertion) => {
-	Assertion.add('hexString', function hexString() {
-		this.params = {
-			operator: 'to be hex string',
-		};
-		Buffer.from(this.obj, 'hex')
-			.toString('hex')
-			.should.equal(this.obj);
-	});
+/* eslint-disable no-underscore-dangle */
+Assertion.addProperty('hexString', function handleAssert() {
+	const actual = this._obj;
 
-	Assertion.add('integer', function integer() {
-		this.params = {
-			operator: 'to be an integer',
-		};
-		parseInt(this.obj, 10).should.equal(this.obj);
-	});
+	new Assertion(actual).to.be.a('string');
+
+	const expected = Buffer.from(actual, 'hex').toString('hex');
+	this.assert(
+		expected === actual,
+		'expected #{this} to be a hexString',
+		'expected #{this} to not be a hexString',
+	);
+});
+
+Assertion.addProperty('integer', function handleAssert() {
```

Comment:
Maybe these `handleAssert` functions should have distinct names?

Reply:
it could be lambda function, but do you have any suggestion for the name?
maybe like `checkInteger` and `checkHexString`?